packages/snaps-utils/src/manifest/validators/package-json-recommended-fields.ts

```diff
@@ -0,0 +1,19 @@
+import type { ValidatorMeta } from '../validator-types';
+
+const RECOMMENDED_FIELDS = ['repository'] as const;
+
+/**
+ * Checks if package.json contains recommended fields.
+ */
+export const packageJsonRecommendedFields: ValidatorMeta = {
+  severity: 'warning',
```

Comment:
Rather than having a severity here, maybe it makes more sense to have different functions on the context object? Then a single rule would be able to do this:

```
context.warn("Missing recommended property.");
context.error("Missing required property.");
```

Reply:
The idea is to have single-purpose rules as much as possible. Your example should be split into two different rules.